example/w-hot-reloading/hot_reloading.ml

```diff
@@ -0,0 +1,89 @@
+let hotreload_script
+    ?(retry_interval_ms = 500)
+    ?(max_retry_ms = 5000)
+    ?(route = "/_hotreload")
+    ()
+  =
+  Printf.sprintf
+    {js|
+var socketUrl = "ws://" + location.host + "%s"
+var s = new WebSocket(socketUrl);
+
+s.onopen = function(even) {
+  console.log("WebSocket connection open.");
+};
+
+s.onclose = function(even) {
+  console.log("WebSocket connection closed.");
+  const innerMs = %i;
+  const maxMs = %i;
+  const maxAttempts = Math.round(maxMs / innerMs);
+  let attempts = 0;
+  function reload() {
+    attempts++;
+    if(attempts > maxAttempts) {
+      console.error("Could not reconnect to dev server.");
+      return;
+    }
+
+    s2 = new WebSocket(socketUrl);
+
+    s2.onerror = function(event) {
+      setTimeout(reload, innerMs);
+    };
+
+    s2.onopen = function(event) {
+      location.reload();
+    };
+  };
+  reload();
+};
+
+s.onerror = function(event) {
+  console.error("WebSocket error observed:", event);
+};
+|js}
+    route
+    retry_interval_ms
+    max_retry_ms
+
+let inject_hotreload_script
+    ?(reload_script = hotreload_script ())
+    ()
+    (next_handler : Dream.request -> Dream.response Lwt.t)
+    (request : Dream.request)
+    : Dream.response Lwt.t
+  =
+  let%lwt response = next_handler request in
+  match Dream.header "Content-Type" response with
+  | Some "text/html" | Some "text/html; charset=utf-8" ->
```

Comment:
Checking if it starts with `text/html` is probably more robust

Reply:
It should probably be a check that it either

- Equals `text/html`.
- Contains `;`, and the text before the first `;` equals `text/html`.